platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/geometry/LatLngQuad.java

```diff
@@ -0,0 +1,95 @@
+package com.mapbox.mapboxsdk.geometry;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+
+import com.mapbox.mapboxsdk.exceptions.InvalidLatLngBoundsException;
+import com.mapbox.services.android.telemetry.constants.GeoConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A geographical area representing a non-aligned rectangle
+ * <p>
+ * This class does not wrap valies to the world bounds
+ * </p>
+ */
+public class LatLngQuad implements Parcelable {
```

Comment:
@1ec5 I looked through SpriteKit, SceneKit, and the Maps APIs for HERE, Tangram, Google, Amazon and Yandex. None of those have a concept of a non-aligned rect. Beyond rects, they all support polygons or arrays of points. 

Im going to add convenience methods to support  `LatLngBounds` and `MGLCoordinateBounds` for the SDKs. This avoids the awkwardness of the `*Quad`/`NonAlignedBounds` names for those who have _regular_ aligned images